packages/aws-cdk-lib/core/lib/removal-policys.ts

```diff
@@ -0,0 +1,171 @@
+
+  /**
+   * If true, overwrite any user-specified removal policy that has been previously set.
+   * This means even if the user has already called `applyRemovalPolicy()` on the resource,
+   * this aspect will override it.
+   * @default false - do not overwrite user-specified policies
+   */
+  readonly overwrite?: boolean;
+}
+
+/**
+ * The RemovalPolicyAspect handles applying a removal policy to resources
+ */
+class RemovalPolicyAspect implements IAspect {
+  constructor(
+    private readonly policy: RemovalPolicy,
+    private readonly props: RemovalPolicyProps = {},
+  ) {}
+
+  /**
+   * Converts pattern (string or class) to resource type string
+   */
+  private convertPatternToResourceType(pattern: string | typeof CfnResource): string {
+    if (typeof pattern === 'string') {
+      return pattern;
+    } else {
+      const cfnType = (pattern as any).CFN_RESOURCE_TYPE_NAME;
+      if (typeof cfnType !== 'string') {
+        throw new Error(`Class ${pattern.name} must have a static CFN_RESOURCE_TYPE_NAME property.`);
+      }
+      return cfnType;
+    }
+  }
+
+  /**
+   * Checks if the given resource type matches any of the patterns
+   */
+  private resourceTypeMatchesPatterns(resourceType: string, patterns?: (string | typeof CfnResource)[]): boolean {
+    if (!patterns || patterns.length === 0) {
+      return false;
+    }
+
+    const resourceTypeStrings = patterns.map(pattern => this.convertPatternToResourceType(pattern));
+    return resourceTypeStrings.includes(resourceType);
+  }
+
+  public visit(node: IConstruct): void {
+    if (!CfnResource.isCfnResource(node)) {
+      return;
+    }
+
+    const cfnResource = node as CfnResource;
+    const resourceType = cfnResource.cfnResourceType;
+
+    const userAlreadySetPolicy = cfnResource.cfnOptions.deletionPolicy !== undefined ||
+                                 cfnResource.cfnOptions.updateReplacePolicy !== undefined;
+
+    if (!this.props.overwrite && userAlreadySetPolicy) {
+      return;
+    }
+
+    if (this.resourceTypeMatchesPatterns(resourceType, this.props.excludeResourceTypes)) {
+      return;
+    }
+
+    if (
+      this.props.applyToResourceTypes?.length &&
+      !this.resourceTypeMatchesPatterns(resourceType, this.props.applyToResourceTypes)
+    ) {
+      return;
+    }
+
+    // Apply the removal policy
+    try {
+      cfnResource.applyRemovalPolicy(this.policy);
+    } catch (error) {
+      // Check if the error is an instance of the built-in Error class
+      if (error instanceof Error) {
+        throw new Error(`Failed to apply removal policy to resource type ${resourceType}: ${error.message}`);
+      } else {
+        // If it's not an Error instance, convert it to a string for the message
+        throw new Error(`Failed to apply removal policy to resource type ${resourceType}: ${String(error)}`);
+      }
+    }
+  }
+}
+
+/**
+ * Manages removal policies for all resources within a construct scope
+ */
+export class RemovalPolicys {
```

Comment:
Is it better 'RemovalPolicies' or others?

Reply:
That's right!

https://github.com/aws/aws-cdk/pull/32283/commits/710c35eb5ad5ce0fd180a35ec480f6a847acea45

Reply:
Could you please change the title to `feat(core): RemovalPolicies...`?

Reply:
Sorry, I fixed the issue too.